File: Python/Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        rl = list(ransomNote)
        ml = list(magazine)
        rl.sort()
        ml.sort()
        
        i = 0
        for ch in rl:
            for i in xrange(i, len(ml)):
                if ch == ml[i]:
                    i += 1
                    break
            else:
                break
        else:
            return True
        return False
            
        

File: cpp/Ransom Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        sort(ransomNote.begin(), ransomNote.end());
        sort(magazine.begin(), magazine.end());
        int idx = 0;
        bool jump_flag = true;
        for (char ch : ransomNote) {
            bool flag = true;
            for (; idx < magazine.size(); ++idx) {
                if (ch == magazine[idx]) {
                    idx++;
                    flag = false;
                    break;
                }
            }
            if (flag) {
                jump_flag = false;
                break;
            }
        }
        return jump_flag;
    }
};
